### PR TITLE
complete `scipy.signal._savitzky_golay`

### PR DESCRIPTION
**Pull request by pavyamsiri**
Before this PR, the stub file `_savitzky_golay.pyi` had untyped annotations. This PR adds in the missing type annotations and improves previous ones.

The improvements are:
1. Use `op.CanIndex` instead of `int` for the `axis` parameter.
2. Use Literals to type hint certain strings for `use` and `mode` parameters

This file tends to use `int` and `float` a lot which could perhaps be replaced by `AnyInt` and `AnyReal` however, some parameters like `window_length` are used as shapes in calls to functions like `np.zeros` which don't accept `np.bool_`.

For simplicity I left the type annotations as the builtin python types but I am open to using more general type annotations to allow numpy dtypes as well. It just might take a bit more testing to check the allowed types.

### CODE REVIEW

File: scipy-stubs/signal/_savitzky_golay.pyi [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
 def savgol_coeffs(
     window_length: int,
     polyorder: int,
     deriv: int = 0,
     delta: float = 1.0,
     pos: int | None = None,
-    use: str = "conv",
-) -> UntypedArray: ...
+    use: Literal["conv", "dot"] = "conv",
+) -> _Array_fc_1d: ...
```
Comment (jorenham) [inline scipy-stubs/signal/_savitzky_golay.pyi:17]: I can't find anything in the docs about complex return types. So I'm missing something, then using `np.floating[Any]` instead here would be best.

Comment (pavyamsiri) [reply scipy-stubs/signal/_savitzky_golay.pyi]: Thanks for the heads up. I think I just saw that the output was due to some sort of convolution and took the return type from there. I don't think the coefficients of the filter can be complex but I'm also not too well versed on filters.

Comment (pavyamsiri) [reply scipy-stubs/signal/_savitzky_golay.pyi]: Should be fixed now.

**Comment by pavyamsiri** [comment]
> Unless you want to change something, I think this we can go ahead and merge this :)

Yes should be all done now.

**Comment by jorenham** [comment]
Thanks again @pavyamsiri !

**Pull request merged**